src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -13,11 +13,13 @@ module Kore.Step.BaseStep
     , StepperVariable (..)
     , StepProof (..)
     , StepProofAtom (..)
+    , UnificationProcedure (..)
     , VariableRenaming (..)
     , simplificationProof
     , stepProof
     , stepProofSumName
     , stepWithAxiom
+    , stepWithAxiom'
```

Comment:
Since you exported this, I'd rename it to something that makes the difference from `stepWithAxiom` more clear if possible, maybe `stepWithAxiomForUnifier`, maybe something else.